vite-css-component.js

```diff
@@ -0,0 +1,68 @@
+import * as path from 'path';
+import * as sass from 'sass';
+
+const files_to_compile = [
+  '/assets/scss/fonts.scss',
+  '/assets/scss/base.scss',
+];
+
+export function cssPerComponentPlugin() {
+  return {
+    name: 'vite-css-component',
+    async transform(code, id) {
+      // Exclude Vue SFC style blocks
+      if (id.endsWith('.scss') && !id.includes('?vue&type=style')) {
+        const isBaseFile = files_to_compile.some(file => id.endsWith(file));
+
+        if (isBaseFile) {
+          // Extract the base file name
+          const baseFileName = path.basename(id, '.scss');
+
+          // Define the output CSS path for base files
+          const outputCssPath = `dist/css/base/${baseFileName}.css`;
```

Comment:
This seems problematic because the path is inherently going to be relative to whatever location this script gets run from. I suspect this might be why the files get placed in `/storybook-static` when the `storybook-build` command gets run.